src/query.js

```diff
@@ -40,22 +55,44 @@ class Query {
     const run = {
       peersSeen: new Set(),
       errors: [],
-      peersToQuery: null
+      paths: null // array of states per disjoint path
     }
 
     if (peers.length === 0) {
       this._log.error('Running query with no peers')
       return callback()
     }
 
-    waterfall([
-      (cb) => PeerQueue.fromKey(this.key, cb),
-      (q, cb) => {
-        run.peersToQuery = q
-        each(peers, (p, cb) => addPeerToQuery(p, this.dht, run, cb), cb)
-      },
-      (cb) => workerQueue(this, run, cb)
-    ], (err) => {
+    // create correct number of paths
+    const numPaths = Math.min(c.DISJOINT_PATHS, peers.length)
+    const pathPeers = []
```

Comment:
```js
const pathPeers = Array(numPaths).fill([])
```

Reply:
And remove the cycle below

Reply:
That won't work, since all of the array elements will be references to the same empty array.